### ISSUE TITLE
colors.currentStep not working on PieChart

### ISSUE THREAD

**Issue by lucianholt97** [amcharts/amcharts4#1282]
It seems like setting `colors.currentStep` is not working for `PieCharts`, yet.
Setting `chart.colors.currentStep = 3` on a `XYChart` works as expected.
According to the docs I need to set the `ColorSet` on a `PieChart` for each series. But `pieSeries.colors.currentStep = 3` does not have any effect. Neither does `pieChart.colors.currentStep = 3.`

Also see: https://codepen.io/lucianholt/pen/eoVbve?=true

Please let me know if I am missing anything.
Thanks in advance!

**Comment by zeroin** [comment]
This happens because when we validate data, we reset `currentStep` to 0 each time (and this also happens on first draw), as number of slices depend on data. Otherwise on each data change color slices would change and I bet nobody wants that. So I am afraid there is no way to solve this except to modify the list of colors in ColorSet (or in theme).

**Comment by lucianholt97** [comment]
@zeroin thanks for your answer. I understand that the `currentStep` has to be reset on data updates. Setting the `currentStep` before adding data should modify the start color step. In my use case, I have multiple `pieCharts` side by side, each with a variable number of slices. In order to differentiate the data, I'd like them to have different colors like it was only one chart. I need to set the default `currentStep` prop on the other pieCharts to achieve this. This should also have been implemented according to #652 and works on `XYCharts` already.

**Comment by martynasma** [comment]
@lucianholt97 It's not possible to compare XYChart and PieChart.

The former has one ColorSet which assigns a new color for each series. It does not have to assign individual colors to each individual data item.

The latter is different in that it can have multiple series and each data item within series needs to have a distinctive color assigned, hence its not possible to reuse same color set for everything.

Furthermore, `currentStep` will not work if set directly on series' `colors`, because colors are assigned to slices based on index, not using `next()`.

We might add support for `currentStep` at some point, but at this time, I don't have what to suggest :(

**Comment by lucianholt97** [comment]
@martynasma thank you for the detailed explanation. The difference makes totally sense. Is there any other workaround to achieve the same result? Would it be possible to manually remove the first x colors from the colorSet and then pass the modified colorSet to the chart? And will this be responsive when I change the colorSet?

Then, I would create the colorSets dynamically from “outside” and pass it to each PieChart component as prop (working with Vue).

**Comment by martynasma** [comment]
This is a bit hacky but works:

```
pieSeries.colors.generate(20);
pieSeries.colors.list.splice(0, 10);
```

This will ensure that slices in series start using from the 10th color.

**Comment by martynasma** [comment]
Fixed in [4.4.9](https://github.com/amcharts/amcharts4/releases/tag/4.4.9).

## [4.4.9] - 2019-05-28

### Added
- `startIndex` property added to `ColorSet`. This allows to specify from which color charts like `PieChart` should start when picking colors for slices.
- `excludeFromTotal` property added to `XYSeries` (default: `false`). Allows excluding values of particular series when caluclating totals.
- `includeRangesInMinMax` property added to `ValueAxis`.

### Changed
- The chart will now error out if trying to instantiate generic classes like  (`Chart`, `Axis`, `Series`).
- Legend in `TreeMap` was completely revamped. Previously it used series as data, but as series were not always created, it failed to render items in some cases. Now it uses `TreeMapDataItem` of `TreeMap`.
- `TreeMap` legend will now show items of first level which has more than one data item.
- `dataContext` of `ForceDirectedNodeDataItem` will now have the same object as target node's.

### Fixed
- `responsive.useDefault = false` was being ignored.
- `DateAxis` with hourly-based data could render some of the labels using date formatter of a changed period even if the period didn't actually change.
- In some cases `MapChart` could interfere with event objects of other charts on the same page, even when not directly interacting with it.
- If `baseInterval` of `DateAxis` didn't have `count` set, the chart failed to render.
- Setting data on hidden Series resulted tooltips on invisible bullets to be shown.
- `WordCloudSeries` with single label or several labels with the same values was displaying labels using `minFontSize`.
- `axis.createSeriesRange()` was not working with `RadarColumnSeries`.
- When data was added to a chart, series could flicker at incorrect position before rendering properly. Also the same could happen when zooming chart (it was only visible if animated theme was not in use).

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
